runner/diff.go

```diff
@@ -7,8 +7,24 @@ import (
 
 func diffState(logger *util.Logger, prevState state.PersistedState, newState state.PersistedState, collectedIntervalSecs uint32) (diffState state.DiffState) {
 	diffState.StatementStats = diffStatements(newState.StatementStats, prevState.StatementStats)
-	diffState.RelationStats = diffRelationStats(newState.RelationStats, prevState.RelationStats)
-	diffState.IndexStats = diffIndexStats(newState.IndexStats, prevState.IndexStats)
+	diffState.DBStats = make(map[state.Oid]*state.DiffedDBStats)
+	for dbOid := range newState.DBStats {
+		newDbStats := newState.DBStats[dbOid]
+		prevDbStats := prevState.DBStats[dbOid]
+		var prevRelStats state.PostgresRelationStatsMap
+		var prevIdxStats state.PostgresIndexStatsMap
+		if prevDbStats != nil {
+			prevRelStats = prevDbStats.RelationStats
+			prevIdxStats = prevDbStats.IndexStats
+		} else {
+			prevRelStats = make(state.PostgresRelationStatsMap)
+			prevIdxStats = make(state.PostgresIndexStatsMap)
```

Comment:
Do we actually need to do this initialization?

My understanding is that its safe to access a key in an uninitialized map: https://play.golang.org/p/TKm5W9UmsIT